### ISSUE TITLE
EVP_PKEY_derive_set_peer difference between 1.1.1 and 3.0

### ISSUE THREAD

**Issue by danbev** [openssl/openssl#14831]
In Node.js we have a few tests that pass in different Diffie-Hellman parameters (like group and generator) and the tests expect an error to be raised in this situation when linked against OpenSSL 1.1.1. 
But when linked against OpenSSL 3.0 this [check](https://github.com/openssl/openssl/blob/b1c908f421b3466aecf980603132bcab89d1ce99/crypto/evp/exchange.c#L406-L410) of different parameters does not seem to exist.

For example, this standalone program (no Node.js code) [dh_1_1_1.c](https://github.com/danbev/learning-libcrypto/blob/master/dh_1_1_1.c) is linked against OpenSSL 1.1.1:
```console
$ ./dh_1_1_1 
Diffie Hellman example
prime size: 192
prime size: 1024
EVP_PKEY_derive_set_peer failed
errno: 101298329, error:0609B099:digital envelope routines:EVP_PKEY_derive_set_peer:different parameters
```
And [dh.c](https://github.com/danbev/learning-libcrypto/blob/master/dh.c) is linked against OpenSSL 3.0 and does not raise an error when run.

My question is should this check also be present in OpenSSL 3.0?

Thanks,

**Comment by slontis** [comment]
Yes you are correct.
See fix in https://github.com/openssl/openssl/pull/14823

**Comment by danbev** [comment]
> See fix in #14823

Oh very nice, thank you!
